### PR TITLE
Fix promise rejection error in cygwin tests

### PR DESCRIPTION
**Pull request by dgolovin**
Fix removes done() call to fix the error.

### CODE REVIEW

**Comment by dgolovin** [comment]
test this

**Comment by jrichter1** [comment]
I don't think this change is going to fix anything. The problem here is that the callback in detectExistingIstall(cb) is in fact swallowing assertion errors. So if the assertion actually fails it only puts the Unhandled promise rejection error into output, but the test still passes.
You can go ahead and try it with some bogus expectations. It just won't make the test fail.
This needs a different approach, maybe we could get rid of the callback and just chain the promise forward?

File: test/unit/model/cygwin-test.js [hunk 1]
```diff
@@ -245,7 +245,6 @@ describe('Cygwin installer', function() {
         installer.detectExistingInstall(function() {
```
Comment (jrichter1) [inline test/unit/model/cygwin-test.js]: This is the call that needs a return statement to be executed properly. I guess there are other in the code as well. Can you take a look?

**Comment by dgolovin** [comment]
if mocha callback executed at the end of detection callback everything seems to be working properly.

**Comment by dgolovin** [comment]
I updated PR with right fix and tested with wrong expectations and it fails, becuase done method is not being called at the end of callback.

**Comment by jrichter1** [comment]
Actually you can remove the done callback/function. The problem was that in order to test an async function you need to call it with return. As in

return asyncFunction(callback() {
  expect(something)
})

**Comment by dgolovin** [comment]
This should work for promises, but obviously does not for what we have now.

**Comment by jrichter1** [comment]
Right, silly me. Is there a particular reason to have the promise chain end with a callback that then get wrapped into a promise again?

**Comment by dgolovin** [comment]
Do you mean pomisify detectExistingInstall()?

**Comment by dgolovin** [comment]
test this

**Comment by jrichter1** [comment]
Yes, technically it should just be about replacing the callback with a resolve/reject and removing the wrapper in confirm controller. (And changing the tests.)

**Comment by dgolovin** [comment]
I'll refactor that. Should we push this one? I will file a jira for refactoring.

**Comment by dgolovin** [comment]
test this

**Comment by dgolovin** [comment]
Opened https://issues.jboss.org/browse/JBDS-4223 to refactor detection merhods to return promise.

**Pull request merged**